concrete-core/src/backends/core/implementation/engines/glwe_secret_key_generation.rs

```diff
@@ -0,0 +1,91 @@
+use crate::backends::core::implementation::engines::CoreEngine;
+use crate::backends::core::implementation::entities::{GlweSecretKey32, GlweSecretKey64};
+use crate::backends::core::private::crypto::secret::GlweSecretKey as ImplGlweSecretKey;
+use crate::specification::engines::{GlweSecretKeyGenerationEngine, GlweSecretKeyGenerationError};
+use concrete_commons::parameters::{GlweDimension, PolynomialSize};
+
+impl GlweSecretKeyGenerationEngine<GlweSecretKey32> for CoreEngine {
+    /// # Example:
+    /// ```
+    /// use concrete_commons::parameters::{GlweDimension, PolynomialSize};
+    /// use concrete_core::prelude::*;
+    /// # use std::error::Error;
+    /// #
+    /// # fn main() -> Result<(), Box<dyn Error>> {
+    /// let mut engine = CoreEngine::new()?;
+    /// // DISCLAIMER: the parameters used here are only for test purpose, and not secure.
+    /// let glwe_dimension = GlweDimension(2);
+    /// let polynomial_size = PolynomialSize(4);
+    /// let glwe_secret_key: GlweSecretKey32 = engine
+    ///     .generate_glwe_secret_key(glwe_dimension, polynomial_size)
+    ///     ?;
+    /// assert_eq!(glwe_secret_key.glwe_dimension(), glwe_dimension);
+    /// assert_eq!(glwe_secret_key.polynomial_size(), polynomial_size);
+    /// engine.destroy(glwe_secret_key)?;
+    /// #
+    /// # Ok(())
+    /// # }
+    /// ```
+    fn generate_glwe_secret_key(
```

Comment:
Hi @Anas-B! For all the engines I did not include a description of what the function does, since they should be described in the specification instead of the backend. @aPere3 wrote the documentation so that you should go through the specification before entering an engine documentation page, if I'm not mistaken, so I didn't even put a link to the specification page. Maybe we should put one anyway?

Reply:
@Anas-B I've just added a description for each trait implementation, that points to the specification.

Reply:
Thanks @Anas-B for the remark. As Agnes mentioned, the usual way to navigate the doc in rust would be to jump from the trait implementation block to the trait declaration by clicking the name of the trait. My opinion on that would be to leave it without any particular documentation (moreover those are hidden by default in the doc). Maybe we should discuss that with someone else ?

Reply:
Discussing this further with Agnes, we noticed that there was another good point to the way it is done currently: It makes clearer which `impl` block does what when viewing all the implementations from the trait definition page. This is very valuable I think, because otherwise, it makes the different implementations difficult to differentiate. I think we can leave that there for the time being.